### DISCUSSION TITLE
RateLimiter perWeek and perMonth methods

### DISCUSSION THREAD

**Discussion by sts-ryan-holton** [laravel/framework#53001] [Ideas]
Hi, I'm implementing a feature into my project that allows a user to provide "deduplication" and "rate-limitng" of leads through the system. We need "weeks" and "months" for this to work, but there's no method on the `Limit` facade for `perWeek` or `perMonth`.

```php
RateLimiter::for($key, function () use ($data) {
    $occourances = intval(Arr::get($data, 'dedupable.value', 0));

    return match (Arr::get($data, 'dedupable.unit', 'minutes')) {
        'seconds' => Limit::perSecond($occourances),
        'hours' => Limit::perHour($occourances),
        'days' => Limit::perDay($occourances),
        'weeks' => null, // Laravel team to add,
        'months' => null, // Laravel team to add
        default => Limit::perMinute($occourances)
    };
});
```

Given that these appear to just be time calculations behind the scenes, may I suggest either introducing some kind of customisation to allow a specific period, or add more methods like:

```
/**
 * Create a new rate limit using days as decay time.
 *
 * @param  int  $maxAttempts
 * @param  int  $decayWeeks
 * @return static
 */
public static function perWeek($maxAttempts, $decayWeeks = 1)
{
    return new static('', $maxAttempts, 60 * 60 * 24 * 7 * $decayWeeks);
}
```

**Comment by henzeb** [comment]
Doesn't Limit implement the Macroable trait ?

**Reply by sts-ryan-holton** [reply]
Not sure what you mean by this?

**Reply by henzeb** [reply]
If it implements Macroable, you can add the methods yourself.

If not, it is not the laravel team's job, you can submit a PR yourself ...
